transpiler/src/target/typescript/core/components/repo/repoPort/helpers/domainIDofRepoPort.ts

```diff
@@ -19,56 +19,45 @@
 // belonging to the Aggregate, and create all the CRUD methods with the respective data types.
-import {
-  TTargetDependenciesTypeScript,
-  TContextData,
-  TBitloopsPrimaryType,
-} from '../../../../../../../types.js';
+import { TTargetDependenciesTypeScript } from '../../../../../../../types.js';
 import {
   BitloopsTypesMapping,
   ClassTypes,
   TClassTypesValues,
 } from '../../../../../../../helpers/mappings.js';
 import { modelToTargetLanguage } from '../../../../modelToTargetLanguage.js';
 import { IntermediateASTTree } from '../../../../../../../ast/core/intermediate-ast/IntermediateASTTree.js';
+import { RootEntityDeclarationNode } from '../../../../../../../ast/core/intermediate-ast/nodes/RootEntity/RootEntityDeclarationNode.js';
 
 export const findIdOfRepoDomainObject = (
   repoDependencyName: string,
-  bitloopsModel: IntermediateASTTree,
+  ast: IntermediateASTTree,
   type: TClassTypesValues,
-  context: TContextData,
 ): TTargetDependenciesTypeScript => {
-  const { boundedContext, module } = context;
   if (type === ClassTypes.RootEntity) {
-    const aggregate = bitloopsModel[boundedContext][module][type][repoDependencyName];
-    if (!aggregate) {
+    const rootEntityNode = ast.getAggregateNodeWithIdentifier(
+      repoDependencyName,
+    ) as RootEntityDeclarationNode;
+    if (!rootEntityNode) {
       throw new Error(`${type} ${repoDependencyName} not found`);
     }
-    const aggregateProps = aggregate.create.parameter.type as string;
-    const propsModel = bitloopsModel[boundedContext][module].Props[aggregateProps];
-    if (!propsModel) {
-      throw new Error(`Props ${aggregateProps} not found`);
-    }
-    const idProp = propsModel.variables.find((prop) => prop.name === 'id');
-    const idType: TBitloopsPrimaryType = idProp.type;
-    const idTypeRes = modelToTargetLanguage({
-      value: { type: idType },
-      type: BitloopsTypesMapping.TBitloopsPrimaryType,
-    });
-    return idTypeRes;
-  } else if (type === ClassTypes.ReadModel) {
-    const readModel = bitloopsModel[boundedContext][module][type][repoDependencyName];
-    if (!readModel) {
-      throw new Error(`${type} ${repoDependencyName} not found`);
+
+    const aggregatePropsNode = ast.getPropsFromEntity(rootEntityNode);
+
+    if (!aggregatePropsNode) {
+      const aggregateNodeIdentifier = rootEntityNode.getIdentifier().getValue();
+      throw new Error(`Props for aggregate ${aggregateNodeIdentifier} not found`);
     }
 
-    const idProp = readModel.variables.find((prop) => prop.name === 'id');
-    const idType: TBitloopsPrimaryType = idProp.type;
+    const propsValue = ast.getValueOfPropsWithIdentifierFromDomainCreate(aggregatePropsNode, 'id');
+
     const idTypeRes = modelToTargetLanguage({
-      value: { type: idType },
+      value: propsValue,
       type: BitloopsTypesMapping.TBitloopsPrimaryType,
     });
     return idTypeRes;
+  } else if (type === ClassTypes.ReadModel) {
```

Comment:
here in case of the ReadModel do we need ids? It seems we don't to me but I don't know if I miss some case.

Reply:
![image](https://user-images.githubusercontent.com/63424729/209135644-30f4a633-a69b-4378-952d-f8a26ff71da7.png)